app/lib/io/zip.js

```diff
@@ -0,0 +1,34 @@
+define([
+    'jszip'
+  , 'io/readDirRecursive'
+], function(
+    JSZip
+  , readDirRecursive
+) {
+    "use strict";
+
+    var unpack = function(zipData, io, targetPath){
+        /* TODO: Implement-me! */
+    };
+
+    var encode = function(io, sourcePath){
+        var zip = new JSZip();
+        var files = readDirRecursive(false, io, sourcePath);
+        var i;
+
+        for (i in files){
+            var file = files[i];
+            var relative_path = file.split(sourcePath)[1];
+            var data = io.readFile(false, file);
```

Comment:
My understanding it that readDirRecursive will return only a list of full filepaths. It will not include the directory entries.